src/static_canvas.class.js

```diff
@@ -459,13 +467,18 @@
     __setBgOverlayImage: function(property, image, callback, options) {
       if (typeof image === 'string') {
         fabric.util.loadImage(image, function(img) {
-          img && (this[property] = new fabric.Image(img, options));
+          if (img) {
+            var instance = new fabric.Image(img, options);
+            this[property] = instance;
+            instance.canvas = this;
+          }
           callback && callback(img);
```

Comment:
Can we return fabric image object i.e. instance, instead of img?

Reply:
That is another method, is called `fabric.Image.fromUrl(...)`

Reply:
Yes but on callback of `setBackgroundImage/ setOverlayImage` it returns image element not fabric.Image instance.

Reply:
Does it even make sense that those callback return something?
You are right they should at least return the same thing.

I would remove the method completely to be honest. Devs should create theyr background object and set it normally.